shared/blob_utils.py

```
@@ -1,50 +1,138 @@
+        try:
+            await container_client.get_container_properties()
+        except ResourceNotFoundError:
+            await service.create_container(BLOB_CONTAINER)
+
+        # Full path to blob
+        blob_path: str = f"{folder.strip('/')}/{filename}"
+        blob_client: BlobClient = container_client.get_blob_client(blob_path)
+
+        # Fetch content
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)